### PR TITLE
Angular2 template

### PR DESCRIPTION
**Pull request by sanfordredlich**
Hi, 

Dan Grove said he'd be happy if I put together an Angular2 example app, so here it is, complete with gnomes. I'll add to it over time, per the roadmap you can see on the example app's [repository](https://github.com/sanfordredlich/angular2-dart-gnome-example-app). 

What do you think, helpful?

I've also put together rough detailed instructions for creating Stagehand templates since I didn't find any. Would they be helpful for me to write up in Markdown, maybe to extend the developers' guide?

Thanks,

- Sanford

### CODE REVIEW

**Comment by sethladd** [comment]
Thanks! Please do add any instructions or notes to this project's Wiki. Much appreciated!

@devoncarew : this might be the perfect reason to land the third-party generators support :)

**Comment by sanfordredlich** [comment]
Cool, will do. Will the third-party plans obviate the need for my instructions, though?

**Comment by sethladd** [comment]
I'm not certain what third-party generators will look like (@devoncarew has some thoughts here) but maybe we should hold off on a developer's guide until we know how to add a third-party gen.

**Comment by sanfordredlich** [comment]
I already added [these instructions](https://github.com/google/stagehand/wiki). You might want to take a look and consider hiding them. 

Do you think this Angular2 app is a good fit then or will the Angular team be making one? I know lots of people are interested in getting something working right now.

**Comment by sethladd** [comment]
I don't know that the Angular team's plans are. Maybe @kevmoo has some
ideas.

We'd love to have a third-party template for Angular2, that's for sure!

**Comment by devoncarew** [comment]
Love it! This would be a great way showcase of third-party templates. Let's keep this PR on hold for ~a week. If we make progress on supporting third-party templates in that time, let's convert the angular sample over to that.

**Comment by sanfordredlich** [comment]
Ok, sounds good, thanks. Looking forward to the third-party code system. Hope my instructions for the main template system come in handy for updates to that. Cheers,  - S.

File: templates/angular2-dart-gnome-example-app/pubspec.yaml [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+version: 0.0.1
+description: Angular2 Dart gnome example app
+#author: {{author}} <email@example.com>
+#homepage: https://www.example.com
+environment:
+  sdk: '^1.9.3'
+dependencies:
+  angular2: "2.0.0-alpha.25"
+  browser: "^0.10.0"
+dev_dependencies:
```
Comment (kwalrath) [inline templates/angular2-dart-gnome-example-app/pubspec.yaml]: Have you tried the angular2 transformer here? If it works, it'd be great to include it. (I'm adding it to the quickstart with https://github.com/angular/angular.io/pull/133.)

**Comment by kwalrath** [comment]
@rkirov might know if the Angular team has any plans for an angular2 template.

**Comment by rkirov** [comment]
No plans that I know of on the dart-side of things, @igorminar has plans for angular cli that would contain a scaffold, so it would be nice to have feature-parity with this template.

**Comment by kwalrath** [comment]
So... Add the angular2 transformer, and update to alpha.26, and it should be good to go.

**Comment by sanfordredlich** [comment]
Sure, I'll add the angular2 transformer as the next step on my roadmap. I can also change the vibe of the app to make it have less gnome and more of a vanilla template for an application, if that will help.

**Comment by sanfordredlich** [comment]
Great, will do. I've already made those changes to the app the template is derived from.

**Comment by devoncarew** [comment]
@sanfordredlich, I sent you a PR against your branch. I don't want to gate getting an angular2 sample w/ getting generalized templates done, and I haven't made progress on those.

I tested your template - it works as is, but the app doesn't run if I update to the latest angular2. Can you update it to the latest angular? Thanks!

**Comment by sethladd** [comment]
I'm concerned that angular2 isn't 1.0 yet. Can we hold off on including an
a2 template until we get third party template support or it goes 1.0?

**Comment by sanfordredlich** [comment]
Sure, I'm happy to wait for Angular2 to be 1.0, whatever you guys would like.

**Comment by kevmoo** [comment]
@sethladd @sanfordredlich @devoncarew 

I think angular 2 is stable enough where we may want to let folks start playing with it.

Could we create a convention for `-beta` templates?

**Comment by kevmoo** [comment]
@sanfordredlich I'm working on an updated version of this. Thanks for the inspiration!

**Comment by sanfordredlich** [comment]
Hey Kevin, cool, do you want help?

**Comment by kevmoo** [comment]
We're live – https://github.com/google/stagehand/commit/cc57dc1aa256c6b00fbc8ddb55df2b55f5c9655f

**Comment by sanfordredlich** [comment]
Cool, thanks for putting that together, Kevin!

**Pull request closed without merging**